sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/arithmetic.scala

```diff
@@ -187,6 +193,12 @@ case class Abs(child: Expression, failOnError: Boolean = SQLConf.get.ansiEnabled
 abstract class BinaryArithmetic extends BinaryOperator with SupportQueryContext {
   override def nullIntolerant: Boolean = true
 
+  override def contextIndependentFoldable: Boolean =
+    left.contextIndependentFoldable && right.contextIndependentFoldable &&
+      !DataTypeUtils.containsTimestampOrUDT(left.dataType) &&
```

Comment:
which `BinaryArithmetic` accepts timestamp?

Reply:
Check https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/BinaryArithmeticWithDatetimeResolver.scala#L61

Reply:
OK, I was trying to be conservative. Let's make it simple and skip the data type check then.